searchings/linear_search.nim

```diff
@@ -0,0 +1,68 @@
+## Linear Search in Pure Nim
+## =============
+#[
+Linear search is the simplest but least efficient algorithm
+to search for an element in a data set.
+It examines each element until it finds a match,
+starting at the beginning of the data set, until the end.
+The search is finished and terminated once the target element is located.
+# https://www.simplilearn.com/tutorials/data-structure-tutorial/linear-search-algorithm
+]#
+## Worst case time complexity: O(N)
+## Average case time complexity: O(N)
+## Best case time complexity: O(1)
+## Space complexity: O(1)
```

Comment:
Not for the recursive algorithm. I think you can assume that you are in O(n) due to the call stack. I do not think there are any [CPS](https://en.wikipedia.org/wiki/Continuation-passing_style) automatic optimization in the Nim compiler.